trees-and-graphs/route-between-nodes.mdx

```diff
@@ -0,0 +1,86 @@
+# Read the solution [Here](https://quastor.org/cracking-the-coding-interview/trees-and-graph/route-between-nodes)
+
+# Route Between Nodes
+
+## Cracking The Coding Interview 4.1
+
+<br/>
+
+# Question
+
+Given a directed graph, design an algorithm to find out whether there is a route between two nodes
+
+Example graph
+
+<img
+  src="./assets/route-between-nodes.png"
+  alt="Directed graph"
+  height="200"
+  width="400"
+  layout="responsive"
+/>
+
+```
+
+Input -  start = A, end = F
+Output - True
+
+Input - start = A, end = C
+Output - False
+
+```
+
+<details>
+  <summary>BFS Solution</summary>
+
+Start a breadth-fist-search from the start node, check if the end node is found. keep track of visited nodes to avoide cycles and repetition of nodes.
+    
+```python
+from queue import Queue
+from collections import defaultdict
+from typing import DefaultDict, List
+
+def search(graph: DefaultDict ,start : str,end : str) -> bool:
```

Comment:
nit: fmt.

you might want to incorporate flake8 or black to autoformat

Reply:
but the core algos look good :)

Reply:
Thanks for the suggestions, it helps.